cobc/parser.y

```diff
@@ -5752,7 +5776,6 @@ collating_sequence_clause:
 	check_repeated ("COLLATING", SYN_CLAUSE_3, &check_duplicate);
 	current_file->collating_sequence = alphanumeric_collation;
 	current_file->collating_sequence_n = national_collation;
-	CB_PENDING ("FILE COLLATING SEQUENCE");
```

Comment:
This should be changed to `CB_UNFINISHED` - it only works for BDB (LMDB has this function internal only, D-ISAM and likely C-ISAM have no option for passing that from the outside (D-ISAM has `ISCUSTKEYS`, but that is only available with licensed versions built from source with custom compare function added), V*ISAM has the extension `issetcollate()` - but so far the passed collation is stored and then never used).

Please also add missing pieces for `collating_sequence_keys` - the `#ifdef 0` can be removed from `cob_file_key` so that you have the matching field (it is only allocated in libcob); the code in fileio.c to set the key into app_data would first check the key, then the file. The most work here is in the compiler and also those should be not that much..